experiments/functional/blockdevice-reusability/test.yml

```diff
@@ -0,0 +1,252 @@
+            executable: /bin/bash
+          register: spc_status
+          until: '"{{ spc_pool_name }}" not in spc_status.stdout'
+          retries: 30
+          delay: 10
+
+        - name: Verify if the CSP is getting removed
+          shell: kubectl get csp -n {{ operator_ns }} -l openebs.io/cstor-pool-cluster={{ spc_pool_name }}
+          args:
+            executable: /bin/bash
+          register: csp_status
+          until: "'No resources found.' in csp_status.stderr"
+          retries: 30
+          delay: 10
+
+        - name: Verify if the cStor pool pods are deleted
+          shell: kubectl get pods -n {{ operator_ns }} -l openebs.io/storage-pool-claim={{ spc_pool_name }}
+          args:
+            executable: /bin/bash
+          register: pool_status
+          until: "'No resources found.' in pool_status.stderr"
+          retries: 30
+          delay: 10
+
+        - name: Verify if the BDC are deleted
+          shell: kubectl get bdc -n {{ operator_ns }} -l openebs.io/storage-pool-claim={{ spc_pool_name }}
+          args:
+            executable: /bin/bash
+          register: bdc_status
+          until: "'No resources found.' in bdc_status.stderr"
+          retries: 30
+          delay: 10
+
+        - name: Verify if the blockDevices are unclaimed
+          shell: >
+            kubectl get blockdevices -n {{ operator_ns }} 
+            -o jsonpath='{.items[?(@.metadata.name=="{{ item }}")].status.claimState}'
+          args:
+            executable: /bin/bash
+          register: bd_status
+          with_items: "{{ blockdevice_name.stdout_lines }}"
+          until: "'Unclaimed' in bd_status.stdout"
+          retries: 30
+          delay: 10
+
+        - name: Create cstor cspc disk pool
+          shell: kubectl apply -f cspc.yml
+          args:
+            executable: /bin/bash
+
+        - name: Check whether cspc is created
+          shell: >
+            kubectl get cspc -n {{ operator_ns }} -o custom-columns=:.metadata.name --no-headers
+          args:
+            executable: /bin/bash
+          register: cspc_name
+          until: "cspc_pool_name in cspc_name.stdout"
+          delay: 30
+          retries: 10
+
+        - name: Verify the status of CSPI
+          shell: >
+            kubectl get cspi -n {{ operator_ns }} -l openebs.io/cstor-pool-cluster={{ cspc_pool_name }}
+            -o custom-columns=:.status.phase --no-headers
+          args:
+            executable: /bin/bash
+          register: cspi_status
+          until: "((cspi_status.stdout_lines|unique)|length) == 1 and 'ONLINE' in cspi_status.stdout"
+          delay: 5
+          retries: 60
+
+        - name: Obtain the CSPI name to verify the status
+          shell: >
+            kubectl get cspi -n {{ operator_ns }} -l openebs.io/cstor-pool-cluster={{ cspc_pool_name }}
+            -o custom-columns=:.metadata.name --no-headers
+          args:
+            executable: /bin/bash
+          register: cspi_name
```

Comment:
can we swap the above two tasks.
- Obtain the cspi list first and then verify its status